### ISSUE TITLE
String.replace() errornous compares and replaces one byte instead of two bytes

### ISSUE THREAD

**Issue by bmarwell** [eclipse-openj9/openj9#19903]
Hello,

given this sample program:

```java
    import java.util.regex.Pattern;
    
    final class Scratch {
      private static final String LEFT_BRACKET = "\u1161";
      private static final Pattern LEFT_BRACKET_PATTERN = Pattern.compile(LEFT_BRACKET, Pattern.LITERAL);
    
       private static final String TEST_STRING = "Max Mustermann / normal Customer / not relevant";
    
      public static void main(final String[] args) {
        final String replaced = TEST_STRING.replaceAll(LEFT_BRACKET, "[");
        // JDK 22 prints: Max Mustermann / normal Customer / not relevant
        // 11.0.17-tem prints: Max Mustermann / normal Customer / not relevant
        // 11.0.18-sem prints: Max Mustermann / normal Customer / not relevant
        // 11.0.22-sem prints: M[x Musterm[nn / norm[l Customer / not relev[nt
        // 11.0.23-sem prints: M[x Musterm[nn / norm[l Customer / not relev[nt
        System.out.println(replaced);
    
        final String replacedWithPattern = LEFT_BRACKET_PATTERN.matcher(replaced).replaceAll("[");
        // JDK 22 prints: Max Mustermann / normal Customer / not relevant
        // 11.0.17-tem prints: Max Mustermann / normal Customer / not relevant
        // 11.0.18-sem prints: Max Mustermann / normal Customer / not relevant
        // 11.0.22-sem prints: M[x Musterm[nn / norm[l Customer / not relev[nt
        // 11.0.23-sem prints: M[x Musterm[nn / norm[l Customer / not relev[nt
        System.out.println(replacedWithPattern);
      }
    }
```

Our developer expects this program to output a copy of the string, but in fact, all "a" characters get replaced. This is highly unexpected and dangerous.

Also confirmed with JDK11 build 20240723-095802 on Mac from OpenJ9 jenkins.

Case number: TS016798420

**Comment by bmarwell** [comment]
Does not occur with this input (added brackets):
```java
TEST_STRING.replaceAll( "[\u1161]", "[" );
```

**Comment by keithc-ca** [comment]
I expect the test intended to start with the same string for both paths:
```java
        final String replacedWithPattern = LEFT_BRACKET_PATTERN.matcher(TEST_STRING).replaceAll("[");
```
The difference is that it shows that `Pattern.Matcher.replaceAll()` behaves as expected.

**Comment by pshipton** [comment]
The fixes to head, 0.46, 0.47 are merged.

**Comment by bmarwell** [comment]
Thank you so much for this quick fix!!! 👍🏻
